toplevel/topdirs.ml

```diff
@@ -185,6 +185,29 @@ let _ = add_directive "mod_use" (Directive_string (with_error_fmt dir_mod_use))
 
 exception Bad_printing_function
 
+let type_arrow ta tb =
+  Ctype.newty (Tarrow (Asttypes.Nolabel, ta, tb, commu_var ()))
+
+let type_formatter () =
+  let format =
+    Env.lookup_module_path
+      ~use:false ~loc:Location.none ~load:false
+      (Longident.Lident "Format")
+      !toplevel_env in
+  Ctype.newconstr (Path.Pdot(format, "formatter")) []
```

Comment:
I would propose to forge the `format` identifier with `Ident.create_persistent`.

Reply:
Ah, yes.

Reply:
Rebased.